### PR TITLE
Saml-attributestatement KB

### PR DESCRIPTION
**Pull request by fuzzmaestro1**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

**Comment by s-santillan** [comment]
FYI @fuzzmaestro1 I'll go ahead and add commits to this PR. [Slack thread for context](https://semgrepinc.slack.com/archives/C02PV1VVAUR/p1699548943278229)

File: docs/kb/semgrep-cloud-platform/saml-attributestatement.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+If you don't know what attribute the payload is failing on, you can investigate the payload to determine the attribute. Refer to [How to view a SAML response in your browser](https://support.okta.com/help/s/article/How-to-View-a-SAML-Response-in-Your-Browser-for-Troubleshooting?language=en_US) for debugging steps.
+
+After finding out what attribute is causing the error, proceed to the following section.
+
+## If you know the SAML attribute that is causing the error
+
+There are two possible solutions for this problem, depending on the expectations of your service provider (SP):
+
+- If your service provider anticipates a value for the specific SAML Attribute statement, you must include a value within the SAML settings. [Where do we include this value? From the service provider?]
```
Comment (s-santillan) [inline docs/kb/semgrep-cloud-platform/saml-attributestatement.md:27]: @fuzzmaestro1 For this line, can you let me know where we're expected to put the Attribute Statement? (I assume it's in the SP)

File: docs/kb/semgrep-cloud-platform/saml-attributestatement.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- Conversely, if the service provider does not expect that specific Attribute statement to be transmitted, remove the Attribute statement from the SAML settings. [Now I assume this is from Semgrep's side?]
```
Comment (s-santillan) [inline docs/kb/semgrep-cloud-platform/saml-attributestatement.md:28]: Can you also let me know if this is on Semgrep's side?

File: docs/kb/semgrep-cloud-platform/saml-attributestatement.md (hunk 1 above)
Comment (fuzzmaestro1) [reply docs/kb/semgrep-cloud-platform/saml-attributestatement.md:27]: @s-santillan The SAML provider, the IdP (not the SP) is responsible for the attributes.  
The service provider (SP)is the Semgrep App, but it is not responsible for the attributes.  The attributes are populated in the Identity Provider (IdP, such as Okta, OneLogin etc).  If the attribute in the IdP is not configured or misconfigured, this is manifested as a kind of AttributeStatement error is thrown by the SP (Semgrep in this case).

**Comment by s-santillan** [comment]
Got it. Thank you!

**Comment by s-santillan** [comment]
Hey @fuzzmaestro1! 

I have here below a screenshot of your first draft:
Snag_126f3d

Am I correct that the SPs in **red** are meant to be Semgrep?

What about the SP in **blue**?

Thank you.

**Comment by s-santillan** [comment]
@khorne3 i think i asked you to take a look at this last week, and you provided the updated text for it already - i'll make the commit. sorry for the confusion.

**Comment by khorne3** [comment]
@s-santillan I took a stab at getting the final details down. Here's the preview:

image

**Review by khorne3 — APPROVED** [review]

**Pull request merged**